src/lib/exec-fetcher.js

```diff
@@ -25,27 +25,6 @@ const getOptions = () => {
   }
 }
 
-const isExe = (mode, gid, uid) => {
-  if (isWindows()) {
-    return true
-  }
-
-  const isGroup = gid ? process.getgid && gid === process.getgid() : true
-  const isUser = uid ? process.getuid && uid === process.getuid() : true
-
-  return Boolean(mode & 0o0001 || (mode & 0o0010 && isGroup) || (mode & 0o0100 && isUser))
-}
-
-const execExist = async (binPath) => {
-  try {
-    const stat = await fs.statAsync(binPath)
-    return stat.isFile() && isExe(stat.mode, stat.gid, stat.uid)
-  } catch (error) {
-    if (error.code === 'ENOENT') return false
```

Comment:
This `ENOENT` check is the reason we use `ignoreErrors: true`. However, this will ignore all errors, not only `ENOENT`.
Without that option, only `EACCESS` is ignored.